packages/js/product-editor/src/components/attribute-control/attribute-table-row.tsx

```diff
@@ -344,50 +351,87 @@ export const AttributeTableRow: React.FC< AttributeTableRowProps > = ( {
 					suggestions={ tokenFieldSuggestions }
 					value={ tokenFieldValues }
 					onChange={ ( nextTokens: ( TokenItem | string )[] ) => {
+						// If there is no attribute, exit.
 						if ( ! attribute ) {
 							return;
 						}
 
-						// Create a new strings array with the new selected tokens,
-						const nextStringTokens =
-							nextTokens.map( tokenItemToString );
-
 						/*
-						 * Create an array with the new tokens to add,
-						 * filtering the new selected ones that are not in the
-						 * suggestions array.
+						 * Pick the new tokens from the nextTokens array.
+						 * (all new tokens are strings).
 						 */
-						const newTokens = nextStringTokens
-							.filter(
-								( t ) => ! tokenFieldSuggestions.includes( t )
-							)
-							.map( stringToTokenItem );
-
-						// Selected Terms to pass to the Form.
-						const selectedTerms = isGlobalAttribute
-							? nextStringTokens
-							: terms?.filter( ( term ) =>
-									nextStringTokens.includes( term.name )
-							  );
-
-						// Call the callback to update the Form terms.
-						onTermsSelect( selectedTerms, index, attribute );
+						const newTokens = nextTokens.filter(
+							( token ): token is string =>
+								typeof token === 'string'
```

Comment:
Oh TypeScript. I love you most of the time.

Reply:
I researched why TypeScript couldn't ensure that the type was a string array since the `typeof token === 'string'` should be enough. The duality of the token type (`TokenItem | string`) doesn't seem to allow it. :-|